consolidate agent instructions into AGENTS.md, add AI policy (#730)
ionos-cloud/cluster-api-provider-proxmox: CONTRIBUTING.md

make lint test verify
```

## AI-Assisted Contributions

All guidelines in this document apply to AI-assisted contributions. The rules in this section are in addition to those.

AI tools are welcome at any stage.

> **Authorship**: The git commit author must be the person responsible for the contribution. Any agent that produced code must be listed as a co-author using Git's `Co-Authored-By:` trailer, which GitHub renders in the contributor graph. The only exception is a commit that solely modifies an agent instructions file (e.g. `AGENTS.md`, `.github/copilot-instructions.md`): the agent may be the author, with the person as co-author.

Author: ```suggestion
**Authorship**: The git commit author must be the person responsible for the contribution. Any agent that produced code must be attributed with a `Co-Authored-By:` trailer. The only exception is a commit that solely modifies an agent instructions file (e.g. `AGENTS.md`) where the agent may be the author and the person a co-author.
```

**Transparency**: AI agents must be clearly identified as such. We don't mind interacting with agents; we do mind agents masquerading as people. Do not have an agent interact on issues or pull requests while presenting as a human contributor.

**Human oversight**: The human contributor is responsible for the entire contribution and must oversee it end to end — code, tests, and description.

**Quality**: A non-draft pull request must represent completed work:

> - Includes relevant tests covering the changes

Author: It already says 'relevant', 'where applicable' is redundant.
